Core/src/org/sleuthkit/autopsy/datasourceprocessors/xry/XRYCallsFileParser.java

```diff
@@ -39,101 +37,198 @@
 
     private static final Logger logger = Logger.getLogger(XRYCallsFileParser.class.getName());
 
+    //Pattern is in reverse due to a Java 8 bug, see calculateSecondsSinceEpoch()
+    //function for more details.
     private static final DateTimeFormatter DATE_TIME_PARSER
-            = DateTimeFormatter.ofPattern("M/d/y h:m:s [a][ z]");
-
-    private static final String INCOMING = "Incoming";
-
-    //All known XRY keys for call reports.
-    private static final Set<String> XRY_KEYS = new HashSet<String>() {
-        {
-            add("tel");
-            add("number");
-            add("call type");
-            add("name (matched)");
-            add("time");
-            add("duration");
-            add("storage");
-            add("index");
+            = DateTimeFormatter.ofPattern("O a h:m:s M/d/y");
+    /**
+     * All of the known XRY keys for call reports.
+     */
+    private static enum XRY_KEY {
+        TEL("tel"),
+        NAME_MATCHED("name (matched)"),
+        TIME("time"),
+        DIRECTION("direction"),
+        CALL_TYPE("call type"),
+        DURATION("duration"),
+        STORAGE("storage"),
+        INDEX("index"),
+        NAME("name"),
+        NUMBER("number");
+        
+        private final String name;
+        XRY_KEY(String name) {
+            this.name = name;
+        }
+        
+        /**
+         * Indicates if the XRY key is a recognized type.
+         * 
+         * @param xryKey
+         * @return 
+         */
+        public static boolean contains(String xryKey) {
+            String normalizedKey = xryKey.trim().toLowerCase();
+            for(XRY_KEY keyChoice : XRY_KEY.values()) {
+                if(keyChoice.name.equals(normalizedKey)) {
+                    return true;
+                }
+            }
+            
+            return false;
+        }
+        
+        /**
+         * Fetches the enum type for the given XRY key.
+         * 
+         * It is assumed that XRY key string is recognized. Otherwise,
+         * an IllegalArgumentException is thrown. Test all membership
+         * with contains() before hand.
+         * 
+         * @param xryKey
+         * @return 
+         */
+        public static XRY_KEY fromName(String xryKey) {
```

Comment:
You might want to use the existing Enum method valueOf. I don't feel strongly about this.

Reply:
I changed this to 'fromDisplayName', indicating the that string being passed in will differ from the enum name.